src/CLI.ts

```diff
@@ -0,0 +1,178 @@
+  Options.withFallbackConfig(Config.boolean("enforceVersion")),
+  Options.withDescription(
+    "Whether or not @since tags for each module export should be required"
+  )
+)
+
+const runExamples = Options.boolean("no-run-examples", {
+  ifPresent: false,
+  negationNames: ["run-examples"]
+}).pipe(
+  Options.withFallbackConfig(Config.boolean("runExamples")),
+  Options.withDescription(
+    "Whether or not to execute examples discovered in the TypeScript source files"
+  )
+)
+
+const exclude = Options.text("exclude").pipe(
+  Options.repeated,
+  Options.withFallbackConfig(Config.array(Config.string("exclude")).pipe(Config.withDefault([]))),
+  Options.withDescription(
+    "An array of glob patterns specifying files that should be excluded from " +
+      "the generated documentation"
+  )
+)
+
+const compilerOptionsSchema = Schema.record(Schema.string, Schema.unknown)
+
+const parseCompilerOptions = Options.file("parse-tsconfig-file", { exists: "yes" }).pipe(
+  Options.withDescription("The TypeScript TSConfig file to use for parsing source files"),
+  Options.orElse(
+    Options.text("parse-compiler-options").pipe(
+      Options.withDescription("The TypeScript compiler options to use for parsing source files"),
+      Options.mapOrFail((options) =>
+        Schema.parseEither(compilerOptionsSchema)(options).pipe(
+          Either.mapLeft(({ errors }) => {
+            const error = HelpDoc.p(
+              `Invalid TypeScript compiler options:\n${TreeFormatter.formatErrors(errors)}`
+            )
+            return ValidationError.invalidValue(error)
+          })
+        )
+      )
+    )
+  ),
+  Options.optional
+)
+
+const examplesCompilerOptions = Options.file("examples-tsconfig-file", { exists: "yes" }).pipe(
+  Options.withDescription("The TypeScript TSConfig file to use for examples"),
+  Options.orElse(
+    Options.text("examples-compiler-options").pipe(
+      Options.withDescription("The TypeScript compiler options to use for examples"),
+      Options.mapOrFail((options) =>
+        Schema.parseEither(compilerOptionsSchema)(options).pipe(
+          Either.mapLeft(({ errors }) => {
+            const error = HelpDoc.p(
+              `Invalid TypeScript compiler options:\n${TreeFormatter.formatErrors(errors)}`
+            )
+            return ValidationError.invalidValue(error)
+          })
+        )
+      )
+    )
+  ),
+  Options.optional
+)
+
+const options = {
+  projectHomepage,
+  srcDir,
+  outDir,
+  theme,
+  enableSearch,
+  enforceDescriptions,
+  enforceExamples,
+  enforceVersion,
+  runExamples,
+  exclude,
+  parseCompilerOptions,
+  examplesCompilerOptions
+}
+
+/** @internal */
+export const docgenCommand = Command.make("docgen", options)
+
+/** @internal */
+export const cli = docgenCommand.pipe(
+  Command.withHandler(() => Core.program),
+  Command.provideEffect(Configuration.Configuration, (args) => Configuration.load(args)),
+  Command.run({
+    name: "docgen",
+    version: "v1.0.0"
```

Comment:
I wonder if we can load the version from `package.json` here, to reflect the actual version of this library

Reply:
@gcanti - Ah yea this was a mistake.

Since we ship the `package.json` alongside `bin.cjs` in the `dist` folder, perhaps we can just load it directly. The alternative is we could do something similar to what the core `effect` library does using scripts to add a `version.ts` file to the library and load from there. 

https://github.com/Effect-TS/effect/blob/main/scripts/version.mjs

Reply:
@gcanti - what do you think about the options above?

Reply:
@IMax153 I'd say what we do in Effect, so there aren't too many different ways to do the same thing around

Reply:
@gcanti - added the same setup as in the Effect repo - lmk what you think.